chicken-sloths/bangazon-cli-sprint2
# Only display products with available quantity

# Description
When a user requests to add a product to an order, the only products displayed are those which have a quantity > the amount of times it has been ordered. If the user selects an item with a quantity of 0, an error is displayed and the product is not added to their cart

## Related Ticket(s)
Fixes #59 

## Problem to Solve
A user is not shown the products that have a quantity of 0 and they are unable to add products with a quantity of 0

## Proposed Changes
 Because the ID's are no longer ordered, the previous ordered list inaccurately linked with products with their ID's and it looked buggy displaying as a list. For now, these are being displayed with a Purchase Key to avoid confusion. I am open to any suggestions on the UI for this..

## Expected Behavior
A user is not able to add products with quantity of 0

## Steps to Test Solution

1. rebuild db
2. ```npm start```
3. Option 2, select any customer from the list
4. Option 5, select product 24 to add to your order
5. Option 5 again, review list of available products, product 24 should no longer be displaying. try to add product 24 anyways, you will be shown an error and it will not be added to db. 

## Testing


## Documentation
